webroot/src/components/Users/UserList/UserList.ts

```diff
@@ -0,0 +1,284 @@
+import { FormInput } from '@models/FormInput/FormInput.model';
+import { SortDirection } from '@store/sorting/sorting.state';
+import { DEFAULT_PAGE, DEFAULT_PAGE_SIZE, PageChangeConfig } from '@store/pagination/pagination.state';
+import { PageExhaustError } from '@store/pagination';
+
+@Component({
+    name: 'UserList',
+    components: {
+        ListContainer,
+        InputSearch,
+        UserRow,
+    }
+})
+class UserList extends mixins(MixinForm) {
+    @Prop({ required: true }) protected listId!: string;
+
+    //
+    // Data
+    //
+    isInitialFetchCompleted = false;
+    prevKey = '';
+    nextKey = '';
+
+    //
+    // Lifecycle
+    //
+    created() {
+        this.initFormInputs();
+    }
+
+    async mounted() {
+        await this.setDefaultSort();
+        await this.setDefaultPaging();
+        await this.fetchListData();
+        this.isInitialFetchCompleted = true;
+    }
+
+    //
+    // Computed
+    //
+    get sortingStore() {
+        return this.$store.state.sorting;
+    }
+
+    get paginationStore() {
+        return this.$store.state.pagination;
+    }
+
+    get userStore(): any {
+        return this.$store.state.user;
+    }
+
+    get usersStore(): any {
+        return this.$store.state.users;
+    }
+
+    get sortOptions(): Array<any> {
+        const options = [
+            // Temp for limited server paging support
+            // { value: 'firstName', name: this.$t('common.firstName') },
+            { value: 'lastName', name: this.$t('common.lastName'), isDefault: true },
+            // { value: 'permissions', name: this.$t('account.permissions') },
+            // { value: 'affiliation', name: this.$t('account.affiliation') },
+            // { value: 'states', name: this.$t('account.states') },
+            { value: 'accountStatus', name: this.$t('account.accountStatus') },
+        ];
+
+        return options;
+    }
+
+    get headerRecord() {
+        const record = {
+            firstName: this.$t('common.firstName'),
+            lastName: this.$t('common.lastName'),
+            permissionsShortDisplay: () => this.$t('account.permissions'),
+            affiliationDisplay: () => this.$t('account.affiliation'),
+            statesDisplay: () => this.$t('account.states'),
+            accountStatusDisplay: () => this.$t('account.accountStatus'),
+        };
+
+        return record;
+    }
+
+    get emptyMessage(): string {
+        return this.$t('account.usersListEmpty');
+    }
+
+    //
+    // Methods
+    //
+    initFormInputs(): void {
+        this.formData = reactive({
+            userSearch: new FormInput({
+                id: 'user-search',
+                name: 'user-search',
+                // label: computed(() => this.$t('account.userSearchLabel')),
+                placeholder: computed(() => this.$t('account.userSearchLabel')),
+                // validation: Joi.string().min(2).messages(this.joiMessages.string),
```

Comment:
what are these two commented out fields for?

Reply:
Removed